contracts/integration/LockboxAdapter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IXERC20} from "../shared/IXERC20/IXERC20.sol";
+import {IXERC20Lockbox} from "../shared/IXERC20/IXERC20Lockbox.sol";
+import {IConnext} from "@connext/interfaces/core/IConnext.sol";
+import {IXReceiver} from "@connext/interfaces/core/IXReceiver.sol";
+
+interface IXERC20Registry {
+  function getXERC20(address erc20) external view returns (address xerc20);
+
+  function getERC20(address xerc20) external view returns (address erc20);
+
+  function getLockbox(address erc20) external view returns (address xerc20);
+}
+
+contract LockboxAdapter is IXReceiver {
+  address immutable connext;
+  address immutable registry;
+
+  // EVENTS
+  event LockBoxWithdrawFailed(bytes _lowLevelData);
+
+  // ERRORS
+  error Forwarder__is__not__Adapter(address sender);
+  error IXERC20Adapter_WithdrawFailed();
+
+  modifier onlyConnext() {
+    require(msg.sender == connext, "Can only be called by Connext");
+    _;
+  }
+
+  constructor(address _connext, address _registry) {
+    connext = _connext;
+    registry = _registry;
```

Comment:
We don't need to validate if the address is zero?

Reply:
We can, But eventually we are going to deploy it, so we can make sure it's correct address, and xRecieve will also not work in case the address(0) has been set to `_connext` or `registry`